src/core/server/integration_tests/saved_objects/migrations/zero_downtime/create_index.test.ts

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import Path from 'path';
+import fs from 'fs/promises';
+import JSON5 from 'json5';
+import { LogRecord } from '@kbn/logging';
+import { createTestServers, type TestElasticsearchUtils } from '@kbn/core-test-helpers-kbn-server';
+import { getKibanaMigratorTestKit } from '../kibana_migrator_test_kit';
+import { delay } from '../test_utils';
+import { getFooType, getBarType } from './base_types.fixtures';
+
+export const logFilePath = Path.join(__dirname, 'create_index.test.log');
+
+describe('ZDT upgrades - running on a fresh cluster', () => {
+  let esServer: TestElasticsearchUtils['es'];
+
+  const startElasticsearch = async () => {
+    const { startES } = createTestServers({
+      adjustTimeout: (t: number) => jest.setTimeout(t),
```

Comment:
Some basic integration tests on the part of the algo implemented in this PR. Will add more for edge cases once we agree on the overall workflow.